microprofile/WFLY-12313_MicroProfile_OpenAPI_1.1.adoc

```diff
@@ -0,0 +1,111 @@
+This spec provides a unified Java API for producing documentation that complies with the https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md[OpenAPI v3 specification].
+
+An MP OpenAPI subsystem exposes an HTTP GET `/openapi` endpoint that returns an OpenAPI v3 compliant document (in YAML or JSON format) containing documentation of the RESTful services of an application.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12313
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+[ ] Engineering
+
+[X] QE
+
+=== Affected Projects or Components
+
+The SmallRye implementation of the MicroProfile OpenAPI specification is already integrated by the following projects:
+
+* https://thorntail.io/[Thorntail] via the https://github.com/thorntail/thorntail/tree/master/fractions/microprofile/microprofile-openapi[microprofile-openapi fraction].
+* https://quarkus.io/[Quarkus] via the https://github.com/quarkusio/quarkus/tree/master/extensions/smallrye-openapi[smallrye-openapi extension].
+
+These integrations will heavily inform, though not dictate, the corresponding integration into WildFly.
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Create an MP OpenAPI galleon layer
+** Should depend on MP Config and JAX-RS (and their dependencies)
+* The microprofile-openapi spec module must be exposed to deployments
+* The `/openapi` endpoint shall be registered relative to the root of the virtual host associated with the deployment.
+** This places a superficial constraint of a single application per host (see subsequent section re: multiple deployments)
+** In the event a web deployment uses a given host for which another deployment has already registered an `/openapi` endpoint, we will:
+... Log a warning.
+... Skip OpenAPI model generation for that deployment.
+* All WildFly/SmallRye specific configuration will be obtained via MP Config using the properties prefixed with `mp.openapi.extensions`.
+** Consequently, the microprofile-openapi subsystem should have an empty management model.
+* Support YAML and JSON for both input (static/pre-generated OpenAPI document) and output formats.
+* MicroProfile OpenAPI TCK must pass
+
+=== Nice-to-Have Requirements
+
+* Optional features of the specification:
+** Multiple deployments:
+*** If a container allows multiple deployments, the specification expects the `/openapi` endpoint to return the logical union of the documented services for all applications in the runtime. See: https://download.eclipse.org/microprofile/microprofile-open-api-1.1.2/microprofile-openapi-spec.html#_multiple_applications
+**** Initially, we will limit multiple deployment support to a single deployment per host.
```

Comment:
Maybe it would be worth to mention that current SRye implementation assumes single deployment case.